pytorch_lightning/profiler/pytorch.py

```diff
@@ -198,33 +300,56 @@ def __init__(
         self._recording_map: Dict[str, record_function] = {}
+        self._start_action_name: Optional[str] = None
+        self._schedule: Optional[ScheduleWrapper] = None
 
-        if self._export_to_chrome and self._path_to_export_trace is None:
-            rank_zero_warn(
-                "The exported trace would be saved locally as `path_to_export_trace` is None."
-                " Note: Each functions will generate its own traced file."
-            )
+        if _KINETO_AVAILABLE:
+            self.__init_kineto__(profiler_kwargs)
 
         if self._sort_by_key not in self.AVAILABLE_SORT_KEYS:
             raise MisconfigurationException(
                 f"Found sort_by_key: {self._sort_by_key}. Should be within {self.AVAILABLE_SORT_KEYS}. "
             )
 
+    def __init_kineto__(self, profiler_kwargs: Any):
+        has_schedule = "schedule" in profiler_kwargs
+        self._has_on_trace_ready = "on_trace_ready" in profiler_kwargs
+
+        schedule = profiler_kwargs.get("schedule", None)
+        if schedule is not None:
+            if not isinstance(schedule, Callable):
+                raise MisconfigurationException(f"Schedule should be a callable. Found: {schedule}")
+            action = schedule(0)
+            if not isinstance(action, ProfilerAction):
+                raise MisconfigurationException(
+                    f"Schedule should return a `torch.profiler.ProfilerAction`. Found: {action}"
+                )
+        schedule = schedule if has_schedule else self._default_schedule()
+        self._schedule = ScheduleWrapper(schedule) if schedule is not None else schedule
+        self._profiler_kwargs["schedule"] = self._schedule
+
+        activities = profiler_kwargs.get("activities", None)
+        self._profiler_kwargs["activities"] = activities or self._default_activities()
+        self._export_to_flame_graph = profiler_kwargs.get("export_to_flame_graph", False)
+        self._metric = profiler_kwargs.get("metric", "self_cpu_time_total")
+        with_stack = profiler_kwargs.get("with_stack", False) or self._export_to_flame_graph
+        self._profiler_kwargs["with_stack"] = with_stack
+
     def __deprecation_check(
         self,
         profiled_functions: Optional[List[str]],
-        record_functions: Optional[List[str]],
-    ) -> List[str]:
+        record_functions: Optional[Set[str]],
+    ) -> Set[str]:
         if record_functions is None:
-            record_functions = []
+            record_functions = set()
 
         if profiled_functions is not None:
             rank_zero_warn(
                 "`PyTorchProfiler.profiled_functions` has been renamed to"
                 " `record_functions` in v1.3 and will be removed in v1.5", DeprecationWarning
             )
             if not record_functions:
-                record_functions += profiled_functions
+                record_functions |= set(profiled_functions)
             else:
                 raise MisconfigurationException(
```

Comment:
But these defaults can be changed, right?

Reply:
Yes.